openmdao/core/system.py

```diff
@@ -788,8 +791,16 @@ def _final_setup(self, comm, setup_mode, force_alloc_complex=False):
         if setup_mode in ('full', 'reconf'):
             self.set_initial_values()
 
+        # Tell all subsystems to record their metadata if they have recorders attached
         for sub in self.system_iter(recurse=True, include_self=True):
-            sub._rec_mgr.record_metadata(sub)
+            if sub.recording_options['record_metadata']:
+                sub._rec_mgr.record_metadata(sub)
```

Comment:
So, doesn't each sub recorder also need to recurse down below it and record subsystem metadata if record_model_metadata is turned on for that sub recorder?

Reply:
Yeah, this starts to get complicated. I had it that way for a while. But it seemed like the primary (only?) use case for this was to do the recursing from the root. 

Any thoughts on that?

Reply:
Well, I'll confess that I don't quite understand the motivation behind some of this. My thought would be to just record metadata for the variables that you are recording in the case recorder, in which case you would just need one flag to turn that on or off.

Reply:
the rule is that if you record system metadata from any level, then you record it from the root down. There is no partial recording of this. You get all or nothing.

Reply:
Okay. The code above doesn't quite do that. It needs to do a nested system_iter so that each sub from every system_iter loops over all systems in the model and sets it to record that metadata too.

    for sub in self.system_iter(recurse=True, include_self=True):
        sub_rec = sub._rec_mgr

        # The system slotted here needs to record metadata at all levels if flag set to True
        if sub.recording_options['record_metadata']:
            for each_system in self.system_iter(recurse=True, include_self=True):
                sub_rec.record_metadata(each_system)

Does this match what we mean here?

Reply:
Justin, so you are saying that if record_model_metadata is True at any level, then record System metadata for all Systems from the root down? Remember record_model_metadata is by default True, so that if you want to prevent recording all System metadata recording, you would have to set that to False for all Systems. I must be missing something!

Even if we make the detault False, then for this code below, because d1 has record_model_metadata set to True, then again, all Systems from root down will be recorded. It seems a little non-intuitive, if that is the case.

```

        prob = SellarProblem()
        prob.setup()

        recorder = SqliteRecorder("cases.sql")
        prob.model.add_recorder(recorder)

        d1 = prob.model.d1
        d1.recording_options['record_model_metadata'] = True
        d1.add_recorder(recorder)

        prob.run_model()
```

Would it be easier if the record_model_metadata option were on the recorder itself?

Reply:
hrm... you would only have to set it to false anywhere you added a recorder. In 99.9% of times that will either be the root or the top `model` system. 

I think its best to just keep it simple. The option is with all the other options. If you ask for metadata anywhere, you get it all. Default to true

Reply:
So in the Seller example, if we only attach a recorder to d1, and leave record_model_metadata as the default of true, you want all metadata from all the systems in the model recorded to the recorder. Right? 

Have to see how to do that. Subsystems don't have pointers to the model.

Reply:
But what you described is what I want. 
if its not workable because of lack of pointers, then I'll accept that you just store all the metadata from that point down.

Reply:
ok. will see what I can do !